### PR TITLE
WIP: Scope/frame stitching

### PR DESCRIPTION
**Pull request by kurtbrose**
~~doesn't work because assignment is not an expression.... maybe walrus operator to the rescue?~~ (this now only tangentially has to do with `S` assignment, and is more about scope.)

### CODE REVIEW

File: glom/core.py [hunk 1]
```diff
@@ -836,6 +847,14 @@ def __getitem__(self, item):
             return t
         return _t_child(self, '[', item)
 
+    '''  DOH! needs walrus operator
```
Comment (mahmoud) [inline glom/core.py]: not sure assignment expressions allow for overloading, either, fwiw.

Comment (mahmoud) [reply glom/core.py]: In light of that, is there any reason to hold onto this comment?

**Comment by kurtbrose** [comment]
I think we no longer need special casing for `S` in assign:

```
        if self.path.startswith(S):
            dest_target = scope.parents
            dest_path = self.path.from_t()
```

**Comment by mahmoud** [comment]
(when this merges this can also remove the special handling of `.startswith(S)` in Assign)

Also `Let` might not need `over` and such.

**Comment by mahmoud** [comment]
@kurtbrose I think the special case for scope assignment may have to stay; the Assign itself gets its own scope, and it's not very useful to assign to that one.

**Comment by kurtbrose** [comment]
nuts -- I pulled out this comment since this is basically what we were talking about yesterday

```
        # TODO: with scope-chaining, path munging should become part
        # of the constituent specs. This should be addressed when
        # Paths go toward being segments/breadcrumbs.
```

**Comment by kurtbrose** [comment]
I had another bright idea which was this way you can see up and down the scope -- the only thing is you need kind of a pointer to the currently executing scope because what if you are dispatching to multiple children and the parent fails between?

**Pull request merged**